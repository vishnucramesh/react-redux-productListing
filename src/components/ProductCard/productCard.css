.container{
    text-align: center;
}

.cover-image{
    height:150px;
}

.text-container
{
    overflow-x: hidden;
}

.inner-text-container
{
    white-space: nowrap;
}

.text-animation{
    animation: slide infinite 3s alternate;
}

.pad-top-24{
    padding-top: 24px;
}

.pad-bottom-90{
    padding-bottom: 90px;
}

.max-w-px-150{
    max-width: 150px;
}

.min-w-px-80{
    min-width: 80px;
}

@keyframes slide {
  0% {
    margin-left: 0px;
  }
  50% {
    margin-left: -50px;
  }
  100% {
    margin-left: -100px;
  }
}