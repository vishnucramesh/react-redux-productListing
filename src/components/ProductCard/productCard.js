import React, { Component } from 'react';
import './productCard.css';
import {ImageUrl} from '../../environment/environment';
export default class ProductCard extends Component {
    

    render() {
      return (
        <div className="overflow-hidden text-grey-lighter">
          <div className="cover-image w-full bg-grey-darker text-center">
            <img className="cover-image w-full max-w-px-150 min-w-px-80" src={ ImageUrl + this.props.image } onError={(e)=>{e.target.onerror = null; e.target.src=ImageUrl+ "placeholder_for_missing_posters.png"}} alt="Sunset in the mountains"/>
          </div>
          <div className="pad-top-24 pad-bottom-90 w-full">
            <div className="text-sm text-container w-full"><div className={this.props.name !== undefined && this.props.name.length>20?"inner-text-container text-animation":"inner-text-container"} >{this.props.name}</div></div>
          </div>
        </div>
      );
    }

}
