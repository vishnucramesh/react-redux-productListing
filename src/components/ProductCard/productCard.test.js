import React from 'react';
import { shallow } from 'enzyme';
import ProductCard from './productCard';

describe('ProductCardComponent', () => {
  it('should render correctly', () => {
    const component = shallow(<ProductCard />);
    expect(component).toMatchSnapshot();
  });

});