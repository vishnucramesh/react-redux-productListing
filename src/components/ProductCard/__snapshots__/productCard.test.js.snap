// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardComponent should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="cover-image w-full bg-grey-darker text-center"
        >
          <img
            alt="Sunset in the mountains"
            className="cover-image w-full max-w-px-150 min-w-px-80"
            onError={[Function]}
            src="https://vishnucramesh.github.io/ProductListing/assets/undefined"
          />
        </div>,
        <div
          className="pad-top-24 pad-bottom-90 w-full"
        >
          <div
            className="text-sm text-container w-full"
          >
            <div
              className="inner-text-container"
            />
          </div>
        </div>,
      ],
      "className": "overflow-hidden text-grey-lighter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Sunset in the mountains"
            className="cover-image w-full max-w-px-150 min-w-px-80"
            onError={[Function]}
            src="https://vishnucramesh.github.io/ProductListing/assets/undefined"
          />,
          "className": "cover-image w-full bg-grey-darker text-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Sunset in the mountains",
            "className": "cover-image w-full max-w-px-150 min-w-px-80",
            "onError": [Function],
            "src": "https://vishnucramesh.github.io/ProductListing/assets/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="text-sm text-container w-full"
          >
            <div
              className="inner-text-container"
            />
          </div>,
          "className": "pad-top-24 pad-bottom-90 w-full",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="inner-text-container"
            />,
            "className": "text-sm text-container w-full",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "inner-text-container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cover-image w-full bg-grey-darker text-center"
          >
            <img
              alt="Sunset in the mountains"
              className="cover-image w-full max-w-px-150 min-w-px-80"
              onError={[Function]}
              src="https://vishnucramesh.github.io/ProductListing/assets/undefined"
            />
          </div>,
          <div
            className="pad-top-24 pad-bottom-90 w-full"
          >
            <div
              className="text-sm text-container w-full"
            >
              <div
                className="inner-text-container"
              />
            </div>
          </div>,
        ],
        "className": "overflow-hidden text-grey-lighter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Sunset in the mountains"
              className="cover-image w-full max-w-px-150 min-w-px-80"
              onError={[Function]}
              src="https://vishnucramesh.github.io/ProductListing/assets/undefined"
            />,
            "className": "cover-image w-full bg-grey-darker text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Sunset in the mountains",
              "className": "cover-image w-full max-w-px-150 min-w-px-80",
              "onError": [Function],
              "src": "https://vishnucramesh.github.io/ProductListing/assets/undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="text-sm text-container w-full"
            >
              <div
                className="inner-text-container"
              />
            </div>,
            "className": "pad-top-24 pad-bottom-90 w-full",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="inner-text-container"
              />,
              "className": "text-sm text-container w-full",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "inner-text-container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
